### ISSUE TITLE
EnbPI in MAPIE

### ISSUE THREAD

**Issue by valeman** [scikit-learn-contrib/MAPIE#474]
What classes of estimators does EnbPI in MAPIE works with?

The tutorial mentions RandomForest, the EnbPI model as such as published in paper is not limited to bagging estimators and it can work with any model. 

Is there a gap in implementation vs the model in the paper?

If so, it would be good to have EnbPI work with any regression model classes including boosted trees (CatBoost/XGBoost/LightGBM) and scikit-learn regressors.

**Comment by thibaultcordier** [comment]
MAPIE works for all sklearn-compatible estimator classes. To this end, EnbPI already works with any regression model classes including boosted trees (CatBoost/XGBoost/LightGBM) and scikit-learn regressors.

The time series tutorial (https://mapie.readthedocs.io/en/latest/examples_regression/4-tutorials/plot_ts-tutorial.html) mentions RandomForest as an illustration but is not restricted to this class. You can experiment by adapting line 158 of the tutorial with one of the estimators you have just mentioned.

```python
from sklearn.linear_model import LinearRegression
model = LinearRegression()
# or
from sklearn.ensemble import AdaBoostRegressor
model = AdaBoostRegressor()
# or
from sklearn.ensemble import AdaBoostRegressor
model = AdaBoostRegressor()
# or
import lightgbm as lgb
model = lgb.sklearn.LGBMRegressor()
# or
from xgboost import XGBRegressor
model = XGBRegressor()
```

In conclusion, MAPIE is not limited to bagging estimators, it can work with any model and there are no gap in the implementation compared with the model presented in the article.

**Comment by valeman** [comment]
Thank you
